### PR TITLE
Add IBMCloud to IsCloudProviderExternal

### PR DESCRIPTION
**Pull request by jonesbr17**
Adds support for IBMCloud as an external cloud provider.

### CODE REVIEW

File: pkg/cloudprovider/external_test.go
Comment (JoelSpeed) [inline pkg/cloudprovider/external_test.go:160]: Since IBM doesn't have an in-tree provider, I would have expected IBMCloud to always report as external rather than depending on a feature gate, WDYT?

Comment (jonesbr17) [reply pkg/cloudprovider/external_test.go:160]: That makes sense to me. I was basing this off of the example of some of the other providers. The code has been updated.

File: pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go
Comment (Fedosin) [inline pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go:192]: it should be "external", right?

**Review by Fedosin — CHANGES_REQUESTED** [review]

File: pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go
Comment (rtheis) [inline pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go:192]: Should we include IBM in `validCloudProviders := sets.NewString("aws", "azure", "gce", "openstack", "vsphere")` above?  It appears that we should.

Comment (Fedosin) [reply pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go:192]: no, we don't need to include IBM there. That list defines what in-tree cloud providers require `--cloud-config` parameter. For external cloud providers we don't need it.

Comment (JoelSpeed) [reply pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go:192]: At the moment, KCMO uses this name to sync the cloud-config configmap into its namespace for platforms that need it. IBMCloud doesn't need this as KCMO doesn't know anything about IBMCloud, so this case should return empty string as Ovirt and Kubevirt do below

Comment (jonesbr17) [reply pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go]: PR updated based on these comments.

File: pkg/cloudprovider/external_test.go
Comment (rtheis) [reply pkg/cloudprovider/external_test.go:160]: The name of this test should be updated since the feature gates are empty.

File: pkg/operator/configobserver/cloudprovider/observe_cloudprovider_test.go
Comment (rtheis) [inline pkg/operator/configobserver/cloudprovider/observe_cloudprovider_test.go:210]: I think this should be empty to align with NonePlatformType.

**Review by rtheis — CHANGES_REQUESTED** [review]

File: pkg/cloudprovider/external_test.go
Comment (jonesbr17) [reply pkg/cloudprovider/external_test.go:160]: Updated

File: pkg/operator/configobserver/cloudprovider/observe_cloudprovider_test.go
Comment (jonesbr17) [reply pkg/operator/configobserver/cloudprovider/observe_cloudprovider_test.go:210]: This still comes back with "external" based on the changes in the cloudprovider package

**Review by rtheis — APPROVED** [review]
/lgtm

**Review by rtheis — APPROVED** [review]

File: pkg/cloudprovider/external_test.go
Comment (JoelSpeed) [inline pkg/cloudprovider/external_test.go:165]: This test is named incorrectly
```suggestion
	}, {
		name: "No FeatureGate, Platform: IBMCloud",
		status: &configv1.PlatformStatus{
			Type: configv1.IBMCloudPlatformType,
		},
		featureGate: nil,
		expected:    true,
```

**Review by JoelSpeed — COMMENTED** [review]
One nit, otherwise LGTM

File: pkg/cloudprovider/external_test.go
Comment (jonesbr17) [reply pkg/cloudprovider/external_test.go]: Updated

**Review by elmiko — COMMENTED** [review]
/lgtm

totally agree with Joel's comment

**Comment by JoelSpeed** [comment]
/hold cancel

The prerequisite work in the CCCMO is now merged. Once this is merged and revendored in the appropriate components we should be good to go on getting a cluster bootstrapped running the IBM CCM

**Comment by JoelSpeed** [comment]
/assign @sttts 

Stefan, I'm pretty sure this is good to merge based on the previous work we've been doing in this area, but if you could give it a quick glance when you have a moment and add the approve label, would be much appreciated

**Pull request merged**
